code/messaging/src/main/java/com/adobe/marketing/mobile/PropositionItem.java

```diff
@@ -38,15 +42,22 @@ class PropositionItem {
     private static final String MESSAGE_CONSEQUENCE_DETAIL_PUBLISHED_DATE = "publishedDate";
     private static final String MESSAGE_CONSEQUENCE_DETAIL_EXPIRY_DATE = "expiryDate";
     private static final String MESSAGE_CONSEQUENCE_DETAIL_METADATA = "meta";
+    private static final String PAYLOAD_ID = "id";
+    private static final String PAYLOAD_DATA = "data";
+    private static final String PAYLOAD_CONTENT = "content";
+    private static final String PAYLOAD_SCHEMA = "schema";
+    private static final String DATA_RULES_KEY = "rules";
+    private static final String DATA_VERSION_KEY = "version";
+    private static final String JSON_CONTENT_SCHEMA_VALUE = "https://ns.adobe.com/personalization/json-content-item";
 
     // Unique proposition identifier
     private final String uniqueId;
     // PropositionItem schema string
     private final String schema;
     // PropositionItem data content e.g. html or plain-text string or string containing image URL, JSON string
     private final String content;
-    // Weak reference to Proposition instance
-    WeakReference<Proposition> proposition;
+    // Soft reference to Proposition instance
+    SoftReference<Proposition> proposition;
```

Comment:
Can we please rename this as it prevents confusion?
proposition -> propositionReference